Examples/PcapPlusPlus-benchmark/benchmark-google.cpp

```diff
@@ -0,0 +1,263 @@
+#include <Packet.h>
+#include <PcapFileDevice.h>
+#include <PcapPlusPlusVersion.h>
+
+#include <EthLayer.h>
+#include <IPv4Layer.h>
+#include <IPv6Layer.h>
+#include <TcpLayer.h>
+#include <UdpLayer.h>
+
+#include <benchmark/benchmark.h>
+
+#include <iostream>
+
+static std::string pcapFileName = "";
+
+static void BM_PcapFileRead(benchmark::State& state)
+{
+	// Open the pcap file for reading
+	pcpp::PcapFileReaderDevice reader(pcapFileName);
+	if (!reader.open())
+	{
+		state.SkipWithError("Cannot open pcap file for reading");
+		return;
+	}
+
+	size_t totalBytes = 0;
+	size_t totalPackets = 0;
+	pcpp::RawPacket rawPacket;
+	for (auto _ : state)
+	{
+		if (!reader.getNextPacket(rawPacket))
+		{
+			// If the rawPacket is empty there should be an error
+			if (totalBytes == 0)
+			{
+				state.SkipWithError("Cannot read packet");
+				break;
+			}
+			else  // If the rawPacket is not empty, it means the file is over
+			{
+				state.PauseTiming();
+				reader.close();
+				reader.open();
```

Comment:
Why do we need to close and re-open the file? 🤔

Reply:
Google benchmark decides runtime length so the pcap file can (or mostly) is shorter than the actual runtime. So, for rewinding I closed and opened the same file again to get more packets until benchmark finishes.